### PR TITLE
fix(docker): During `cfn submit` docker commands were not working on Windows

### LINKED ISSUE #100: Docker build not supported on windows: module 'os' has no attribute 'geteuid'
I didn't see not supporting Windows called out anywhere in the readme

Repro steps
install:
```
pip install cloudformation-cli
pip install git+https://github.com/aws-cloudformation/aws-cloudformation-rpdk-python-plugin.git#egg=cloudformation-cli-python-plugin
```
init:
```
cfn init
Initializing new project
What's the name of your resource type?
(Organization::Service::Resource)
>> Windows::Bug::Repro
Select a language for code generation:
[1] python36
[2] python37
(enter an integer):
>> 2
Use docker for platform-independent packaging (Y/n)?
This is highly recommended unless you are experienced
with cross-platform Python packaging.
>> y
Initialized a new project in C:\temp\cfn-resource-repro

cp ..\cloudformation-cli-python-lib-0.0.1.tar.gz .\
```
```
cfn submit --dry-run
Starting Docker build. This may take several minutes if the image 'lambci/lambda:build-python3.7' needs to be pulled first.
=== Unhandled exception ===
Please report this issue to the team.
Issue tracker: https://github.com/aws-cloudformation/aws-cloudformation-rpdk/issues
Please include the log file 'rpdk.log'
```
```
[2020-06-15T15:52:09Z] DEBUG    - Logging set up successfully
[2020-06-15T15:52:09Z] DEBUG    - Running submit: Namespace(command=<function submit at 0x000001CD92DE19D0>, dry_run=True, endpoint_url=None, region=None, role_arn=None, set_default=False, subparser_name='submit', use_role=True, verbose=0, version=False)
[2020-06-15T15:52:09Z] DEBUG    - Root directory: C:\temp\cfn-resource-repro
[2020-06-15T15:52:09Z] DEBUG    - Loading project file 'C:\temp\cfn-resource-repro\.rpdk-config'
[2020-06-15T15:52:09Z] INFO     - Validating your resource specification...
[2020-06-15T15:52:09Z] DEBUG    - Resolved '#/definitions/InitechDateFormat' to 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/InitechDateFormat'
[2020-06-15T15:52:09Z] DEBUG    - Parsed 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/InitechDateFormat' to '(<BASE>, 'definitions', 'InitechDateFormat')'
[2020-06-15T15:52:09Z] DEBUG    - Edge from '(<BASE>, 'properties', 'DueDate')' to '(<BASE>, 'definitions', 'InitechDateFormat')'
[2020-06-15T15:52:09Z] DEBUG    - Resolved '#/definitions/InitechDateFormat' to 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/InitechDateFormat'
[2020-06-15T15:52:09Z] DEBUG    - Parsed 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/InitechDateFormat' to '(<BASE>, 'definitions', 'InitechDateFormat')'
[2020-06-15T15:52:09Z] DEBUG    - Edge from '(<BASE>, 'properties', 'ApprovalDate')' to '(<BASE>, 'definitions', 'InitechDateFormat')'
[2020-06-15T15:52:09Z] DEBUG    - Resolved '#/definitions/Memo' to 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/Memo'
[2020-06-15T15:52:09Z] DEBUG    - Parsed 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/Memo' to '(<BASE>, 'definitions', 'Memo')'
[2020-06-15T15:52:09Z] DEBUG    - Edge from '(<BASE>, 'properties', 'Memo')' to '(<BASE>, 'definitions', 'Memo')'
[2020-06-15T15:52:09Z] DEBUG    - Resolved '#/definitions/Memo' to 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/Memo'
[2020-06-15T15:52:09Z] DEBUG    - Parsed 'file:///C:/temp/cfn-resource-repro/windows-bug-repro.json#/definitions/Memo' to '(<BASE>, 'definitions', 'Memo')'
[2020-06-15T15:52:09Z] DEBUG    - Edge from '(<BASE>, 'properties', 'SecondCopyOfMemo')' to '(<BASE>, 'definitions', 'Memo')'
[2020-06-15T15:52:09Z] DEBUG    - Rewriting refs in '<BASE>' (file:///C:/temp/cfn-resource-repro/windows-bug-repro.json)[2020-06-15T15:52:09Z] DEBUG    -   '#/definitions/InitechDateFormat' -> '#/definitions/InitechDateFormat'
[2020-06-15T15:52:09Z] DEBUG    -   '#/definitions/InitechDateFormat' -> '#/definitions/InitechDateFormat'
[2020-06-15T15:52:09Z] DEBUG    -   '#/definitions/Memo' -> '#/definitions/Memo'
[2020-06-15T15:52:09Z] DEBUG    -   '#/definitions/Memo' -> '#/definitions/Memo'
[2020-06-15T15:52:09Z] DEBUG    - overrides.json not found. Not writing to package.
[2020-06-15T15:52:09Z] DEBUG    - Package started
[2020-06-15T15:52:09Z] DEBUG    - 'C:\temp\cfn-resource-repro\build' not found, skipping removal
Traceback (most recent call last):
  File "c:\python38\lib\site-packages\rpdk\python\codegen.py", line 192, in _remove_build_artifacts
    shutil.rmtree(deps_path)
  File "c:\python38\lib\shutil.py", line 737, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\python38\lib\shutil.py", line 596, in _rmtree_unsafe
    onerror(os.scandir, path, sys.exc_info())
  File "c:\python38\lib\shutil.py", line 593, in _rmtree_unsafe
    with os.scandir(path) as scandir_it:
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\temp\\cfn-resource-repro\\build'
[2020-06-15T15:52:09Z] DEBUG    - Dependencies build started from 'C:\temp\cfn-resource-repro'
[2020-06-15T15:52:09Z] DEBUG    - command is 'pip install --no-cache-dir --no-color --disable-pip-version-check --upgrade --find-links /project --requirement /project/requirements.txt --target /project/build'
[2020-06-15T15:52:09Z] WARNING  - Starting Docker build. This may take several minutes if the image 'lambci/lambda:build-python3.7' needs to be pulled first.
[2020-06-15T15:52:09Z] DEBUG    - Trying paths: ['C:\\Users\\user\\.docker\\config.json', 'C:\\Users\\user\\.dockercfg']
[2020-06-15T15:52:09Z] DEBUG    - Found file at path: C:\Users\user\.docker\config.json
[2020-06-15T15:52:09Z] DEBUG    - Found 'auths' section
[2020-06-15T15:52:09Z] DEBUG    - Auth data for xxxxxxxxxxx.dkr.ecr.us-west-1.amazonaws.com is absent. Client might be using a credentials store instead.
[2020-06-15T15:52:09Z] DEBUG    - Found 'credsStore' section
[2020-06-15T15:52:09Z] DEBUG    - Unhandled exception
Traceback (most recent call last):
  File "c:\python38\lib\site-packages\rpdk\core\cli.py", line 98, in main
    args.command(args)
  File "c:\python38\lib\site-packages\rpdk\core\submit.py", line 15, in submit
    project.submit(
  File "c:\python38\lib\site-packages\rpdk\core\project.py", line 336, in submit
    self._plugin.package(self, zip_file)
  File "c:\python38\lib\site-packages\rpdk\python\codegen.py", line 180, in package
    self._build(project.root)
  File "c:\python38\lib\site-packages\rpdk\python\codegen.py", line 199, in _build
    self._docker_build(base_path)
  File "c:\python38\lib\site-packages\rpdk\python\codegen.py", line 256, in _docker_build
    user=f"{os.geteuid()}:{os.getgid()}",
AttributeError: module 'os' has no attribute 'geteuid'
```

Problem line: https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/20f46577f44393a2b50bb0356f825681680035e6/python/rpdk/python/codegen.py#L256

### PR DESCRIPTION
**Pull request by mmaeng**
Closes #100

Docker command to run container was getting the local user euid and gid.  These attributes do not exist on non-UNIX-like OSes like Windows.

Put in check for the command to complete successfully but if not to default to root:root.  Windows ignores these attributes and the Docker file system has built in safeguards around this.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kddejong — APPROVED** [review]

File: python/rpdk/python/codegen.py [hunk 1]
```diff
@@ -281,6 +298,19 @@ def _docker_build(cls, external_path):
             "image '%s' needs to be pulled first.",
             image,
         )
+
+        # Docker will mount the path specified in the volumes variable in the container
+        # and pip will place all the dependent packages inside the volumes/build path.
+        # codegen will need access to this directory during package()
+        # Try to get current effective user ID and Group ID. Valid on UNIX-like systems
+        try:
```
Comment (ericzbeard) [inline python/rpdk/python/codegen.py:306]: Is there a way to detect this without a try? Would be better if we didn't rely on an exception for making the decision.

Comment (mmaeng) [reply python/rpdk/python/codegen.py:306]: There was at least one report of geteuid not working on a new version of MacOS so checking for OS doesn't look to guarantee that this function exists on an OS.  While catching an exception isn't ideal I thought that seemed the most future proof, e.g. when python supports that function on that OS our code doesn't have to change.

Comment (mmaeng) [reply python/rpdk/python/codegen.py:305]: Updated to check with windows before attempting geteuid() for the edge case of the call still failing I added a LOG.warning that it is defaulting to root:root

**Comment by mmaeng** [comment]
In testing branch was mistakenly merged in my fork.  Will redo

**Pull request closed without merging**